Jenkinsfile

```diff
@@ -28,7 +28,11 @@
 
 pipeline {
     agent {
-        label infra.getAgentLabels()
+        docker {
+            label 'docker-dev-1'
+            image infra.getDockerAgentImage()
+            args '--oom-score-adj=100 -v /etc/ssh/id_rsa_git-proxy-cache:/etc/ssh/id_rsa_git-proxy-cache:ro -v /etc/ssh/ssh_config:/etc/ssh/ssh_config:ro -v /etc/gitconfig:/etc/gitconfig:ro'
+        }
```

Comment:
I believe the lines above belong in the shared library's implementation of `infra.getAgentLabels()` and have no business being in the internet-published codebase ;) beside not being portable to whoever else would want to cooperate on the FOSS side and have the code "as is" getting built in their infrastructure.

Reply:
To really fix this, we'd need to completely rewrite the Jenkinsfile to fit the pipeline library, at which point we might as well migrate fty-core to CMake because it would be the last zproject holdout in 42ITy. As for CI portability, without a public pipeline library implementation to run our Jenkinsfile no one but Eaton can run a CI on 42ITy's repositories anyway.

Reply:
Right, in my earlier posting I somehow missed the elephant in the room, that one use-case was `label somethingL()` and another was `docker {somethingD}` (so a workable solution would be to pick between generations of complete `pipeline{}` blocks, as mentioned elsewhere).

I've got a hard time in experiment even passing a variable that could be hidden and abstracted in the lib for the `somethingD` payload. Probably the 2 strings can be moved that way, but not the whole closure (neither docker nor agent).

Probably the smarter solution might be doable with real scripted pipeline, but needs to much hacking if at all possible for declarative :(

So, reluctantly accepting this to unblock our builds; up to you if you'd want to move the docker config strings into the lib.